editor/src/clj/editor/pipeline/bob.clj

```diff
@@ -180,92 +131,148 @@
-  (let [prev-out System/out
+  (let [;; bob might notify the progress tracker AFTER it's done!
+        render-progress! (progress/until-done render-progress!)
+        provided-evaluation-context (some? evaluation-context)
+        evaluation-context (or evaluation-context (g/make-evaluation-context))
+        prev-out System/out
         prev-err System/err
+        log-output-stream (or log-output-stream
+                              (WriterOutputStream. ^PrintWriter *out* StandardCharsets/UTF_8 1024 true))
         ;; Don't close the writer because we don't own the log output stream,
         ;; hence we must not close it. Instead, we will only flush it in the end
         log-stream-writer (PrintWriter. log-output-stream true StandardCharsets/UTF_8)]
-    (with-open [build-out (PrintStream-on
-                            #(doseq [line (util/split-lines %)]
-                               (.println log-stream-writer line)))
-                build-err (PrintStream-on
-                            #(doseq [line (util/split-lines %)]
-                               (.println log-stream-writer line)))]
-      (log/info :bob-command (bob-command-line bob-commands bob-args build-server-headers))
-      (try
-        (System/setOut build-out)
-        (System/setErr build-err)
-        (if (and (some #(= "build" %) bob-commands)
-                 (native-extensions/has-engine-extensions? project evaluation-context)
-                 (not (native-extensions/supported-platform? (get bob-args "platform"))))
-          {:error {:causes (engine-build-errors/unsupported-platform-error-causes project evaluation-context)}}
-          (let [ws (project/workspace project evaluation-context)
-                proj-path (str (workspace/project-path ws evaluation-context))
-                bob-project (Project. java/class-loader (DefaultFileSystem.) proj-path "build/default")]
-            (doseq [[key val] bob-args]
-              (.setOption bob-project key val))
-            (when-not (string/blank? build-server-headers)
-              (doseq [header (string/split-lines build-server-headers)]
-                (.addBuildServerHeader bob-project header)))
-            (.setOption bob-project "liveupdate" (.option bob-project "liveupdate" "no"))
-            (doseq [pkg ["com.dynamo.bob" "com.dynamo.bob.pipeline"]]
-              (ClassLoaderScanner/scanClassLoader java/class-loader pkg))
-            (let [deps (workspace/dependencies ws)]
-              (when (seq deps)
-                (.setLibUrls bob-project (map #(.toURL ^URI %) deps))
-                (ui/with-progress [render-progress! render-progress!]
-                  (.resolveLibUrls bob-project (->progress render-progress! task-cancelled?)))))
-            (.mount bob-project (->graph-resource-scanner ws))
-            (ui/with-progress [render-progress! render-progress!]
-              (run-commands! project evaluation-context bob-project bob-commands render-progress! task-cancelled?))))
```

Comment:
All this code is replaced with a `Bob/invoke` call that does the same. Caveats: 
1. we need to add `resolve` to a list of commands so that Bob does the `(.resolveLibUrls bob-project ...)` call.
2. we no longer ensure `(native-extensions/supported-platform? (get bob-args "platform"))`: this code is basically outdated because Bob also does the same, but better — the editor's hardcoded platforms don't include console plugins.